openshift-kube-apiserver/admission/customresourcevalidation/ingress/validate_ingress.go

```diff
@@ -0,0 +1,156 @@
+package ingress
+
+import (
+	"fmt"
+	"io"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/route/routeapihelpers"
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+const PluginName = "config.openshift.io/ValidateIngress"
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				configv1.Resource("Ingresses"): true,
+			},
+			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+				configv1.GroupVersion.WithKind("Ingress"): featureGateV1{},
+			})
+	})
+}
+
+func toIngressV1(uncastObj runtime.Object) (*configv1.Ingress, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
+	}
+
+	allErrs := field.ErrorList{}
+
+	obj, ok := uncastObj.(*configv1.Ingress)
+	if !ok {
+		return nil, append(allErrs,
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"Ingress"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{"config.openshift.io/v1"}))
+	}
+
+	return obj, nil
+}
+
+type featureGateV1 struct {
+}
+
+var knownFeatureSets = sets.NewString(
+	"",
+	string(configv1.TechPreviewNoUpgrade),
+	string(configv1.CustomNoUpgrade),
+	string(configv1.IPv6DualStackNoUpgrade),
+	string(configv1.LatencySensitive),
+)
+
+func validateIngressSpecCreate(spec configv1.IngressSpec) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// on create, we use the tightest validation
+	for i, currRoute := range spec.ComponentRoutes {
+		if errs := routeapihelpers.ValidateHost(
+			string(currRoute.Hostname),
+			"",
+			field.NewPath("spec.componentRoutes").Index(i).Child("hostname"),
+		); len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		}
```

Comment:
The length check is superfluous.  
```suggestion
		allErrs = append(allErrs, routeapihelpers.ValidateHost(
			string(currRoute.Hostname),
			"",
			field.NewPath("spec.componentRoutes").Index(i).Child("hostname"),
		)...)
```